src/storage/engines/FirebaseStorageEngine.ts

```diff
@@ -376,6 +376,30 @@ export class FirebaseStorageEngine extends StorageEngine {
     return allAnswersPresent;
   }
 
+  async getUserManagementData(key:string) {
+    // Get the user-management collection in Firestore
+    const userManagementCollection = collection(this.firestore, 'user-management');
+    // Grabs all user-management data and returns data based on key
+    const userManagementDocs = await getDocs(userManagementCollection)
+      .then((querySnapshot) => {
```

Comment:
You're using async/await so you shouldn't use then. You should await the thing that you need to wait on and use the result in the logic after